diff --git a/cmake/LibraryUtils.cmake b/cmake/LibraryUtils.cmake
index 396b373819..4bdc9d7bae 100644
--- a/cmake/LibraryUtils.cmake
+++ b/cmake/LibraryUtils.cmake
@@ -71,6 +71,8 @@ function(get_lib_type LIBRARY OUTPUT_VARIABLE)
 			set(LIB_TYPE SHARED)
 		elseif(${LIB_NAME} MATCHES ".*${CMAKE_STATIC_LIBRARY_SUFFIX}")
 			set(LIB_TYPE STATIC)
+		elseif(${LIB_NAME} MATCHES ".*lib")
+			set(LIB_TYPE STATIC)
 		else()
 			message(FATAL_ERROR "Could not determine whether \"${LIBRARY}\" is a static or shared library, it does not have a known suffix.")
 		endif()
