From 8027f55651f15c760d0cae7921bc16104743f2a9 Mon Sep 17 00:00:00 2001
From: Simon Bray <sbray1371@gmail.com>
Date: Fri, 18 Oct 2019 21:56:18 +0200
Subject: [PATCH] patch commit

---
 AmberTools/src/configure2 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/AmberTools/src/configure2 b/AmberTools/src/configure2
index ad4a541..85d30ab 100755
--- a/AmberTools/src/configure2
+++ b/AmberTools/src/configure2
@@ -519,7 +519,7 @@ readline='readline/libreadline.a'
 pic="-fPIC"
 pmemd_openmp='no'
 cpptrajcxxflags=''
-ldflags=''
+ldflags='$LDFLAGS'
 build_sanderapi='build_sanderapi'
 moft='yes'
 #------------------------------------------------------------------------------
-- 
2.17.1

