From 7330111e04424b1fd185b7375a71912a11e58a99 Mon Sep 17 00:00:00 2001
From: Simon Bray <sbray1371@gmail.com>
Date: Sat, 19 Oct 2019 12:36:42 +0200
Subject: [PATCH 1/2] r

---
 AmberTools/src/configure2 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/AmberTools/src/configure2 b/AmberTools/src/configure2
index 797380b..32470b5 100644
--- a/AmberTools/src/configure2
+++ b/AmberTools/src/configure2
@@ -469,7 +469,7 @@ installtype='serial'
 intel_compiler_flag_mkl='no'
 intelmpi='no'
 is_mac='no'
-ldflags=''
+ldflags='$LDFLAGS'
 ld='ld '
 lfs='yes'
 lm='-lm'
-- 
2.17.1


From a8ba9ba188d5a5574c2d9b6e0a37463cf874a3b0 Mon Sep 17 00:00:00 2001
From: Simon Bray <sbray1371@gmail.com>
Date: Sat, 19 Oct 2019 13:19:28 +0200
Subject: [PATCH 2/2] q

---
 AmberTools/src/configure2 | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/AmberTools/src/configure2 b/AmberTools/src/configure2
index 32470b5..3ed06f8 100644
--- a/AmberTools/src/configure2
+++ b/AmberTools/src/configure2
@@ -408,11 +408,11 @@ EOF
 
 # Test for the presence of /bin/csh, since it's needed for installation
 
-if [ ! -x /bin/csh ]; then
-    echo "Error: /bin/csh not found on your system! Install csh or tcsh and rerun"
-    echo "       configure."
-    exit 1
-fi
+# if [ ! -x /bin/csh ]; then
+#    echo "Error: /bin/csh not found on your system! Install csh or tcsh and rerun"
+#    echo "       configure."
+#    exit 1
+#fi
 
 #------------------------------------------------------------------------------
 # Process commandline configuration options:
-- 
2.17.1

