diff --git a/AmberTools/src/configure2 b/AmberTools/src/configure2
index 797380b5..8f071b8b 100755
--- a/AmberTools/src/configure2
+++ b/AmberTools/src/configure2
@@ -3087,7 +3087,8 @@ if [ "$has_fftw3" = 'yes' ]; then
     else
       cd fftw-3.3 && \
         ./configure --prefix=$amberprefix --libdir=$amberprefix/lib \
-        --enable-static $enable_mpi $mpicc $enable_debug $enable_sse\
+        --enable-static $enable_mpi $mpicc $enable_debug \
+        --build=ppc64-unknown-linux \
         CC="$cc" CFLAGS="$cflags $coptflags" \
         F77="$fc" FFLAGS="$fflags $foptflags" \
         FLIBS="$flibs_arch" \
@@ -3158,7 +3159,7 @@ fi
 if [ "$cuda" = "no" ]; then
   printf "\nConfiguring boost (may be time-consuming)..."
   cd boost
-  ./bootstrap.sh --prefix=$amberprefix > ../boost_config.log
+  ./bootstrap.sh --prefix=$amberprefix --with-toolset=gcc > ../boost_config.log
   ncerror=$?
   if [ $ncerror -gt 0 ]; then
       printf "\nError: Boost bootstrap returned $ncerror\n"
