diff --git a/AmberTools/src/quick/CMakeLists.txt b/AmberTools/src/quick/CMakeLists.txt
index 99056204..42b5766c 100644
--- a/AmberTools/src/quick/CMakeLists.txt
+++ b/AmberTools/src/quick/CMakeLists.txt
@@ -51,9 +51,6 @@ set_property(DIRECTORY . PROPERTY COMPILE_DEFINITIONS $<$<CONFIG:Debug>:DEBUG>)
 
 option(WARNINGS "Enable warnings." FALSE)
 
-set(CMAKE_Fortran_FLAGS "")
-set(CMAKE_C_FLAGS "")
-set(CMAKE_CXX_FLAGS "")
 
 set(OPT_FFLAGS "")
 set(OPT_CFLAGS "")
