diff --git a/AmberTools/src/pysander/setup.py b/AmberTools/src/pysander/setup.py
index 580c108a..45f420b0 100644
--- a/AmberTools/src/pysander/setup.py
+++ b/AmberTools/src/pysander/setup.py
@@ -78,6 +78,8 @@ pysander = Extension('sander.pysander',
                      include_dirs=incdir, library_dirs=libdir,
                      libraries=['sander'],
                      depends=depends,
+                     extra_compile_args=['-arch arm64'],
+                     extra_link_args=['-arch arm64'],
                      define_macros=definitions
 )
 
@@ -86,6 +88,8 @@ pysanderles = Extension('sanderles.pysander',
                         include_dirs=incdir, library_dirs=libdir,
                         libraries=['sanderles'],
                         depends=lesdepends,
+                        extra_compile_args=['-arch arm64'],
+                        extra_link_args=['-arch arm64'],
                         define_macros=lesdefinitions)
 
 setup(name='sander',
diff --git a/AmberTools/src/pytraj/setup.py b/AmberTools/src/pytraj/setup.py
index 41cc79a2..319b487a 100644
--- a/AmberTools/src/pytraj/setup.py
+++ b/AmberTools/src/pytraj/setup.py
@@ -111,8 +111,8 @@ if sys.platform.startswith('win'):
     extra_compile_args = []
     extra_link_args = []
 else:
-    extra_compile_args = ['-O0', '-ggdb']
-    extra_link_args = ['-O0', '-ggdb']
+    extra_compile_args = ['-O0', '-ggdb', '-arch arm64']
+    extra_link_args = ['-O0', '-ggdb', '-arch arm64']
 
 
 cython_directives = {
@@ -168,7 +168,7 @@ def prepare_env_for_osx():
         swvers = subprocess.run("sw_vers -productVersion".split(), capture_output=True)
         os.environ["MACOSX_DEPLOYMENT_TARGET"] = ".".join(swvers.stdout.decode("utf-8").strip().split(".")[:2])
 
-        
+
 if sys.platform.startswith('darwin') and is_clang(os.getenv('CXX')):
     prepare_env_for_osx()
     # We don't need this anymore?
