diff --git a/cmake/CompilerFlags.cmake b/cmake/CompilerFlags.cmake
index 914eddbb..b2ed5cf5 100644
--- a/cmake/CompilerFlags.cmake
+++ b/cmake/CompilerFlags.cmake
@@ -215,7 +215,7 @@ if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STR
 	if(TARGET_OSX)
 		# on OS X, Python will link pytraj's extension modules to libc++, so cpptraj needs to use the same standard library
 		add_flags(CXX -stdlib=libc++)
-		set(CMAKE_OSX_DEPLOYMENT_TARGET 10.7)
+		# set(CMAKE_OSX_DEPLOYMENT_TARGET 10.7)
 	endif()	
 	
 	if(OPENMP AND (${CMAKE_CXX_COMPILER_VERSION} VERSION_LESS 3.7))
