diff --git a/AmberTools/src/arpack/Makefile b/AmberTools/src/arpack/Makefile
index fb571b33..3b9ddc1b 100644
--- a/AmberTools/src/arpack/Makefile
+++ b/AmberTools/src/arpack/Makefile
@@ -11,7 +11,8 @@ include ../config.h
 	@echo "[ARPACK]  CC $<"
 	$(VB)$(CC) -c $(COPTFLAGS) $(CFLAGS) -o $@ $<
 
-install: $(LIBDIR)/libarpack.a
+install:
+	@echo "[ARPACK] Skipping compilation..."
 
 ARPACKLIB= ivout.o dvout.o dgetv0.o dlaqrb.o dnaitr.o dnapps.o \
 dnaup2.o dnaupd.o dnconv.o dneigh.o dmout.o \
