diff --git a/cmake/PythonInterpreterConfig.cmake b/cmake/PythonInterpreterConfig.cmake
index 57843ab1..94300e9e 100644
--- a/cmake/PythonInterpreterConfig.cmake
+++ b/cmake/PythonInterpreterConfig.cmake
@@ -86,8 +86,8 @@ To change the Python interpreter in use to a different one inside Anaconda, set
 		set(CMAKE_INCLUDE_PATH_NO_ANACONDA ${CMAKE_INCLUDE_PATH})
 		set(CMAKE_PROGRAM_PATH_NO_ANACONDA ${CMAKE_PROGRAM_PATH})
 		
-		list(APPEND CMAKE_LIBRARY_PATH "${ANACONDA_ROOT}/lib")
-		list(APPEND CMAKE_INCLUDE_PATH "${ANACONDA_ROOT}/include")
+		# list(APPEND CMAKE_LIBRARY_PATH "${ANACONDA_ROOT}/lib")
+		# list(APPEND CMAKE_INCLUDE_PATH "${ANACONDA_ROOT}/include")
 		list(APPEND CMAKE_PROGRAM_PATH "${ANACONDA_BIN}")
 		
 		set(CMAKE_LIBRARY_PATH_ANACONDA ${CMAKE_LIBRARY_PATH})
