diff --git a/CMakeLists.txt b/CMakeLists.txt
index ec8a7d22..02807b35 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -203,7 +203,8 @@ if(BUILD_PYTHON AND DOWNLOAD_MINICONDA)
 endif()
 
 # Generate install rule to bundle and fix OS X dependencies
-add_subdirectory(cmake/BundleOSXDependencies)
+# # DO NOT DO THIS FOR CONDA FORGE PACKAGING
+# add_subdirectory(cmake/BundleOSXDependencies)
 
 #---------------------------------------------------------------------------------------------------------------------------------------------------------------------
 
