diff --git a/AmberTools/src/etc/mdout2pymbar.pl b/AmberTools/src/etc/mdout2pymbar.pl
index c7a93c4bbb..197f49834c 100755
--- a/AmberTools/src/etc/mdout2pymbar.pl
+++ b/AmberTools/src/etc/mdout2pymbar.pl
@@ -1,7 +1,9 @@
-#!/usr/bin/perl -w
+#!/usr/bin/env perl
 # Amber output to PyMBAR input conversion script. Written by Thomas Steinbrecher 05032010
 # Use at your own risk
 
+use warnings;
+
 $calorie = 4.184;
 
 open FILE, $ARGV[0];
diff --git a/AmberTools/src/etc/process_mdout.perl b/AmberTools/src/etc/process_mdout.perl
index b696e2dae4..f0b4140ef1 100755
--- a/AmberTools/src/etc/process_mdout.perl
+++ b/AmberTools/src/etc/process_mdout.perl
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 
 $#ARGV >= 0 || die "usage: $0 mdout_filename [more_mdout_filenames...] \n";
diff --git a/AmberTools/src/etc/process_minout.perl b/AmberTools/src/etc/process_minout.perl
index 0ffa134044..e0ff1b0ac6 100755
--- a/AmberTools/src/etc/process_minout.perl
+++ b/AmberTools/src/etc/process_minout.perl
@@ -1,6 +1,6 @@
-#!/usr/bin/perl -w
+#!/usr/bin/env perl
 
-use strict;
+use strict; use warnings;
 my $ARGV;
 my $filein;
 my $i;
diff --git a/AmberTools/src/etc/sgldinfo.sh b/AmberTools/src/etc/sgldinfo.sh
index c9950d8c05..a29d9b2e97 100755
--- a/AmberTools/src/etc/sgldinfo.sh
+++ b/AmberTools/src/etc/sgldinfo.sh
@@ -1,4 +1,4 @@
-#!/bin/csh
+#!/usr/bin/env csh
 # This script print SGLD properties from sander simulation output
 # csh
 if ($#argv != 1 )then
diff --git a/AmberTools/src/etc/sgldwt.sh b/AmberTools/src/etc/sgldwt.sh
index 2c674f7bbe..a6647ac259 100755
--- a/AmberTools/src/etc/sgldwt.sh
+++ b/AmberTools/src/etc/sgldwt.sh
@@ -1,4 +1,4 @@
-#!/bin/csh
+#!/usr/bin/env csh
 # This script calculate SGLD weight from sander simulation output
 # csh
 if ($#argv != 1 &&$#argv != 2 &&$#argv != 3 )then
diff --git a/AmberTools/src/mm_pbsa/mm_pbsa_statistics.pl b/AmberTools/src/mm_pbsa/mm_pbsa_statistics.pl
index f76e1a2b38..cb310f42a3 100755
--- a/AmberTools/src/mm_pbsa/mm_pbsa_statistics.pl
+++ b/AmberTools/src/mm_pbsa/mm_pbsa_statistics.pl
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!/usr/bin/env perl
 #
 # Script to calc statistics on mm_pbsa output
 #
@@ -7,7 +7,7 @@
 # Last update: 14.02.2014
 #
 
-use strict;
+use strict; use warnings;
 use lib ("$ENV{AMBERHOME}/AmberTools/src/mm_pbsa");
 use mm_pbsa_statistics;
 use mm_pbsa_global qw($HTMLPATH);
diff --git a/AmberTools/src/nmr_aux/senergy b/AmberTools/src/nmr_aux/senergy
index ef7663f18a..79bb3ebbcc 100755
--- a/AmberTools/src/nmr_aux/senergy
+++ b/AmberTools/src/nmr_aux/senergy
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 # Randal R. Ketchem, Ph.D.                619.784.8754 (voice1)
 # Department of Molecular Biology, MB-9   619.784.9879 (voice2)
diff --git a/AmberTools/src/nmr_aux/sviol b/AmberTools/src/nmr_aux/sviol
index 501a4b090b..c9afa718f8 100755
--- a/AmberTools/src/nmr_aux/sviol
+++ b/AmberTools/src/nmr_aux/sviol
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 #
 #   Script to parse sander output restraint analyses, and print a nice
diff --git a/AmberTools/src/nmr_aux/sviol2 b/AmberTools/src/nmr_aux/sviol2
index 006e5f280e..5b4375cf60 100755
--- a/AmberTools/src/nmr_aux/sviol2
+++ b/AmberTools/src/nmr_aux/sviol2
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 if ($ARGV[0] eq "-help" or $#ARGV < 1) { die "
 USAGE:   $0 [d/t] sander_output_files
