diff --git a/AmberTools/src/etc/ceinutil.py b/AmberTools/src/etc/ceinutil.py
index 295baa3e..6bb48aaa 100755
--- a/AmberTools/src/etc/ceinutil.py
+++ b/AmberTools/src/etc/ceinutil.py
@@ -173,6 +173,7 @@ def main(opt):
     # Filter titratable residues based on min and max pKa
     new_reslist = []
     for res in titratable_residues:
+        if getattr(residues, res).Eo is None: continue
         if getattr(residues, res).Eo < mineo: continue
         if getattr(residues, res).Eo > maxeo: continue
         new_reslist.append(res)
