diff --git a/test/rism3d/1ahoa/Run.1ahoa.pme b/test/rism3d/1ahoa/Run.1ahoa.pme
index d4757f16..98e67efc 100755
--- a/test/rism3d/1ahoa/Run.1ahoa.pme
+++ b/test/rism3d/1ahoa/Run.1ahoa.pme
@@ -6,7 +6,7 @@
 
 echo "SANDER: 3D-RISM: PME potential - orthorhombic"
 
-if( ! $?TESTsander ) set TESTsander = "../../../bin/sander"
+if( ! $?TESTsander ) set TESTsander = "${AMBERHOME}/bin/sander"
 
 if ! { ../checkrismunsupported.sh $TESTsander } exit 0
 
