diff --git a/AmberTools/src/etc/cpinutil.py b/AmberTools/src/etc/cpinutil.py
index c7216a8b..7d870abe 100755
--- a/AmberTools/src/etc/cpinutil.py
+++ b/AmberTools/src/etc/cpinutil.py
@@ -188,6 +188,7 @@ def main(opt):
     # Filter titratable residues based on min and max pKa
     new_reslist = []
     for res in titratable_residues:
+        if getattr(residues, res).pKa is None: continue
         if getattr(residues, res).pKa < minpka: continue
         if getattr(residues, res).pKa > maxpka: continue
         new_reslist.append(res)
